core/lib/spree/preferences/preferable.rb

```diff
@@ -158,8 +158,7 @@ def convert_preference_value(value, type)
           value.to_i
         when :boolean
           if !value ||
-             value == 0 ||
-             value =~ /\A(f|false|0)\Z/i ||
+             value.to_s =~ /\A(f|false|0|^)\Z/i ||
```

Comment:
Does it make sense to use this helper instead?
```ruby
ActiveModel::Type::Boolean.new.cast(value)
```
Looking at the [documentation](https://api.rubyonrails.org/classes/ActiveModel/Type/Boolean.html), it seems to do what we need.

Reply:
Yes, I prefer that. It's arguably a breaking change though and one that might be weird to use a config option to control. Thoughts?

Reply:
Hmm, yeah, we may decide to either hold off on this until 3.1 or merge the current version now and migrate to ActiveModel's API in 3.1.

@solidusio/core-team any other thoughts?

Reply:
I'm fine with merging as is, though I'm curious why you'd want to make the switch the the `ActiveModel` API in 3.1 rather than 3.0. Wouldn't that be a breaking change that would be better off rolling into 3.0?

Reply:
Sorry, that comment didn't make any sense. 😅 I meant 3.0, not 3.1. The problem is that ideally we don't want any changes at all in 3.0: we should just remove all deprecations introduced in previous versions of Solidus. The idea is that, if you're on 2.11 and not seeing any deprecations, you can safely upgrade to 3.0.

I think one approach could be to print a deprecation warning if the value of a configuration option is one of the values that is now being treated as true, but would be treated as false by ActiveModel. Something like the following:

```ruby
if value.in?(ActiveRecord::Type::Boolean::FALSE_VALUES - ['f', 'false', '0'])
  Spree::Deprecation.warn "#{value.inspect} will be treated as false starting in Solidus 3.0."
end

# ...
```
The check would have to be a bit more complex, since it looks like we currently treat any object that responds to `#empty?` and returns `true` as false, while ActiveRecord only seems to do it for empty strings, and even then, they return `nil` rather than `false`.

Reply:
Yeah, I agree; that would be ideal. I think we're getting away from the original intent of this PR, though. The goal here was to address a deprecation where `true` is being passed into this class which isn't actually handled by the above deprecation suggestion.

It's more work then I'm sure @jacquesporveau had intended to do here, but maybe we want to leave the existing logic, but wrap it in a conditional deprecation warning that's fired if the existing logic doesn't produce the same value as the `ActiveModel` version (potentially with a config option to use the new `ActiveModel` logic instead.) In v3.0 we remove the old logic and move forward with only the `ActiveModel` version.

@jacquesporveau Apologies for all the back and forth on this one. We're trying to be careful in deciding what belongs in v2.11 since it's our last chance to deprecate things ahead of Solidus v3.0.

Reply:
Sa'll good